java/semantickernel-core/src/main/java/com/microsoft/semantickernel/orchestration/NativeSKFunction.java

```diff
@@ -229,6 +235,11 @@ private static SKNativeTask<SKContext> getFunction(Method method, Object instanc
 
                 return mono.map(
                         it -> {
+                            if (it instanceof Iterable) {
+                                // Handle return from things like Mono<List<?>>
+                                // from {{function 'input'}} as part of the prompt.
+                                it = ((Iterable<?>) it).iterator().next();
```

Comment:
Think taking only the first element from an iterable may be non-obvious behaviour for users

Reply:
Thanks for reminding me. I meant to file an issue for this and will do straight away. 

The other workaround, at least to fit into the current context API, was to change `context.update((String) it)` to `context.update(String.valueOf(it))`, but that would be very strange indeed. Ultimately, I think the fix needs to come in SKContext itself.

Reply:
https://github.com/microsoft/semantic-kernel/issues/2190